src/BinOp.chpl

```diff
@@ -1044,4 +1045,585 @@ module BinOp
     omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
     return new MsgTuple(errorMsg, MsgType.ERROR);
   }
+
+  proc doBigIntBinOpvv(l, r, op: string) throws {
+    var max_bits = max(l.max_bits, r.max_bits);
+    var max_size = 1:bigint;
+    var has_max_bits = max_bits != -1;
+    if has_max_bits {
+      max_size <<= max_bits;
+    }
+    ref la = l.a;
+    ref ra = r.a;
+    var tmp = makeDistArray(la.size, bigint);
+    // i think tmp should be initialized to 0 but just to be sure
+    tmp = 0:bigint;
+    // these cases are not mutually exclusive,
+    // so we have a flag to track if tmp is ever populated
+    var visted = false;
+
+    // had to create bigint specific BinOp procs which return
+    // the distributed array because we need it at SymEntry creation time
+    if l.etype == bigint && r.etype == bigint {
+      // first we try the ops that only work with
+      // both being bigint
+      select op {
+        when "&" {
+          tmp = la & ra;
+          visted = true;
+        }
+        when "|" {
+          tmp = la | ra;
+          visted = true;
+        }
+        when "^" {
+          tmp = la ^ ra;
+          visted = true;
+        }
+        when "/" {
+          tmp = la / ra;
+          visted = true;
+        }
+      }
+    }
+    if l.etype == bigint && (r.etype == bigint || r.etype == int || r.etype == uint) {
+      // then we try the ops that only work with a
+      // left hand side of bigint
+      if r.etype != bigint {
+        // can't shift a bigint by a bigint
+        select op {
+          when "<<" {
+            tmp = la << ra;
+            visted = true;
+          }
+          when ">>" {
+            // workaround for right shift until chapel issue #21206
+            // makes it into a release, eventually we can just do
+            // tmp = la >> ra;
+            var divideBy = makeDistArray(la.size, bigint);
+            divideBy = 1:bigint;
+            divideBy <<= ra;
+            tmp = la / divideBy;
+            visted = true;
+          }
+          when "<<<" {
+            if !has_max_bits {
+              throw new Error("Must set max_bits to rotl");
+            }
+            // return (la << ra) | (la >> (max_bits - ra));
```

Comment:
Is this line a placeholder until Chapel issue #21206 is resolved? If so can we mark it and add it to the issue mentioned in my previous comment? If not, should it be removed?